qiskit/visualization/circuit/qcstyle.py

```diff
@@ -96,9 +96,9 @@ def update(self, other):
         nested_attrs = {"displaycolor", "displaytext"}
         for attr in nested_attrs.intersection(other.keys()):
             if attr in self.keys():
-                self[attr].update(other.pop(attr))
+                self[attr].update(other[attr])
             else:
-                self[attr] = other.pop(attr)
+                self[attr] = other[attr]
 
         super().update(other)
```

Comment:
We have to ensure that the keys in `nested_attrs` are not overwritten again (I think this causes the test failure). We can call `update` without these keys:
```suggestion
        super().update((key, value) for key, value in other.items if key not in nested_attrs)
```